app/v1/connectors/RetrieveTransactionDetailsConnector.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.connectors
+
+import config.AppConfig
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import v1.models.request.retrieveTransactionDetails.RetrieveTransactionDetailsParsedRequest
+import v1.models.response.retrieveTransactionDetails.RetrieveTransactionDetailsResponse
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveTransactionDetailsConnector @Inject()(val http: HttpClient,
+                                                    val appConfig: AppConfig) extends BaseDesConnector {
+
+  def retrieveDetails(request: RetrieveTransactionDetailsParsedRequest)(implicit hc: HeaderCarrier,
```

Comment:
I would prefer if this was either 'retrieveTransactionDetails' or just 'retrieve' as the full name of the allocations endpoint is 'retrieve a self assessment payment's allocation details' so this could potentially be ambiguous.

Reply:
That will be in another connector class, I don't see any confusion. I don't mind changing it, but didn't see any ambiguity.

Reply:
It's up to you. I just thought the naming could be a bit clearer